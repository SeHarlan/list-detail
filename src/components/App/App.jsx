import React, { useState, useEffect } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import List from '../presentational/List/List';
import DetailContainer from '../Container/DetailContainer/DetailContainer';
import { fetchCharacters } from '../../services/avatar';
import ListContainer from '../Container/ListContainer/ListContainer';


export default function App() {
  const [characters, setCharacters] = useState([]);
  const [page, setPage] = useState(1);

  useEffect(() => {
    fetchCharacters(page)
      .then(characters => setCharacters(characters));
  }, [page]);

  const changePage = (direction) => {
    setPage((prevPage) => prevPage + direction);
  };
  return (
    <Router>
      <Switch>
        <Route path="/characters/:id">
          <DetailContainer characters={characters} />
        </Route>
        <Route path="/">
          <ListContainer page={page} characters={characters} changePage={changePage} />
        </Route>

      </Switch>
    </Router>
  );
}
  
