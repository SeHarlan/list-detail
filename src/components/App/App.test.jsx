import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App component', () => {
  const wrapper = shallow(<App />);
  it('renders App', () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it('changes page #', () => {
  //   wrapper.changePage(1);
  //   expect(wrapper.state('page')).toEqual(2);
  // });
});
  
