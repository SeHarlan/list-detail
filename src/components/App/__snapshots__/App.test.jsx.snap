// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      path="/characters/:id"
    >
      <DetailContainer
        characters={Array []}
      />
    </Route>
    <Route
      path="/"
    >
      <ListContainer
        changePage={[Function]}
        characters={Array []}
        page={1}
      />
    </Route>
  </Switch>
</BrowserRouter>
`;
